samples/matter/common/src/diagnostic/diagnostic_logs_provider.h

```diff
@@ -20,8 +14,38 @@
 namespace Nrf::Matter
 {
 
+class DiagnosticLogsIntentImpl {
+public:
+	/**
+	 * @brief Get the captured logs size.
+	 *
+	 * @return size of captured logs in bytes
+	 */
+	virtual size_t GetLogsSize() = 0;
+
+	/**
+	 * @brief Get the captured logs.
+	 *
+	 * @param outBuffer output buffer to store the logs
+	 * @param outIsEndOfLog flag informing whether the stored data is complete log or one of the few chunks
+	 *
+	 * @return size of captured logs in bytes
+	 */
+	virtual CHIP_ERROR GetLogs(chip::MutableByteSpan &outBuffer, bool &outIsEndOfLog) = 0;
+
+	/**
+	 * @brief Finish the log capturing. This method is used to perform potential clean up after session.
+	 */
+	virtual CHIP_ERROR FinishLogs() = 0;
+};
+
 class DiagnosticLogProvider : public chip::app::Clusters::DiagnosticLogs::DiagnosticLogsProviderDelegate {
 public:
+	struct IntentData {
```

Comment:
Theoretically the `IntentData` does not match FiniteMap requirements: `* T2 must have move semantics and bool()/==operators implemented`. Also, if we implement the proper destructor here, we can do without `DeleteIntentImplementation` because the `FiniteMap::Erase()` will do the job then. You can use the following as a reference:
https://github.com/nrfconnect/sdk-nrf/blob/main/samples/matter/common/src/persistent_storage/backends/persistent_storage_secure.h#L36

Reply:
Done